client/src/app/core/services/csv-export.service.ts

```diff
@@ -0,0 +1,128 @@
+import { BaseViewModel } from '../../site/base/base-view-model';
+import { Injectable } from '@angular/core';
+import { FileExportService } from './file-export.service';
+import { TranslateService } from '@ngx-translate/core';
+
+/**
+ * Handles all incoming and outgoing notify messages via {@link WebsocketService}.
+ */
+@Injectable({
+    providedIn: 'root'
+})
+export class CsvExportService {
+    /**
+     * Constructor
+     */
+    public constructor(protected exporter: FileExportService, private translate: TranslateService) {}
+
+    /**
+     * Saves an array of model data to a CSV.
+     * @param data Array of Model instances to be saved
+     * @param columns Column definitions
+     * @param filename name of the resulting file
+     * @param options optional:
+     *      lineSeparator (defaults to \r\n windows style line separator),
+     *      columnseparator defaults to semicolon (other usual separators are ',' '\T' (tab), ' 'whitespace)
+     */
+    public saveAsCsv<T extends BaseViewModel>(
+        data: T[],
+        columns: {
+            property: keyof T;
+            label?: string;
+        }[],
+        filename: string,
+        { lineSeparator = '\r\n', columnSeparator = ';' }: { lineSeparator?: string; columnSeparator?: string } = {}
+    ): void {
+        const all_lines = []; // Array of arrays of entries
+        const used_columns = []; // mapped properties to be included
+
+        // initial array of usable text separators. The first character not used
+        // in any text data or as column separator will be used as text separator
+        let ts_list = ['"', "'", '`', '/', '\\', ';', '.', ','];
+
+        if (lineSeparator === columnSeparator) {
+            throw new Error('lineseparator and columnseparator must differ from each other');
+        }
+
+        ts_list = this.checkCsvTextSafety(lineSeparator, ts_list);
+        ts_list = this.checkCsvTextSafety(columnSeparator, ts_list);
+
+        // create header data
+        const header = [];
+        columns.forEach(column => {
+            const rawLabel = column.label ? column.label : column.property;
```

Comment:
then declare `rawLabel` as string and cast the property: `column.property as string`.